### ISSUE TITLE
init clustering with "User" specified doesn't work if user supplied clusters are not consecutive integers

### ISSUE THREAD

**Issue by kieranrcampbell** [camlab-bioml/starling#44]
When `utility.init_clustering` is called with `initial_clustering_method == "User"` assumes that the user supplied clusters in `labels` are consecutive integers, which then breaks at

`init_e[c, :] = adata.X[init_l == c].mean(0)`

if e.g. `init_l` is a string. 

The solution isn't as simple as casting `labels` to an `int` as no guarantee labels are either integers or consecutive.

Would recommend something like
```python
cluster_idx_map = zip(range(unique_labels), unique_labels)
```

then doing
```python
adata.X[init_l == cluster_idx_map[c]]
```

**Comment by SarahAsbury** [comment]
Also ran into issues if using 1-indexed integers instead of 0-indexed integers for clusters

**Comment by cklamann** [comment]
Closed with #45. @SarahAsbury if your problem persists, please open a separate issue. Thanks!

**Comment by joaolsf** [comment]
Hi, I am running into this issue, whether I use clusters in labels as integers or as category. See below:

ValueError                                Traceback (most recent call last)
Cell In[32], line 1
----> 1 adata = utility.init_clustering('User', ad.read_h5ad('2_h5ad/Pf_spleen_whole_cell_v2'), labels = 'leiden_1')

File ~/Library/Python/3.9/lib/python/site-packages/starling/utility.py:106, in init_clustering(initial_clustering_method, adata, k, labels)
    104     init_ev = np.zeros((k, adata.X.shape[1]))
    105     for i, c in enumerate(classes):
--> 106         init_e[i, :] = adata.X[init_l == c].mean(0)
    107         init_ev[i, :] = adata.X[init_l == c].var(0)
    109 elif initial_clustering_method == "FS":
    110     ## needs to output to csv first
    111     # ofn = OPATH + "fs_" + ONAME + ".csv"

ValueError: could not broadcast input array from shape (94799,44) into shape (44,)

Could you please advise what to do?

Thanks!
